apps/charterafrica/src/payload/utils/mapPlaylistLinkToId.js

```diff
@@ -0,0 +1,16 @@
+export const mapPlaylistLinkToId = ({ siblingData }) => {
+  const { playlistLink } = siblingData;
+  const searchParams = playlistLink?.split("?")?.[1];
+  const playlistId = new URLSearchParams(searchParams).get("list");
```

Comment:
Or you can try creating a URL from link and read the [`searchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams) property. This way you may be able to even detect invalid URLs and show validation error in the CMS.

Reply:
Cool.

```js
  const url = new URL(playlistLink);
  const playlistId = url.searchParams.get("list");
```

No need for `split`, etc. Plus, you can wrap the `new URL` part in a `try`/`catch` to handle invalid URLs.

Reply:
Yes